pkg/obfuscator/domain.go

```diff
@@ -0,0 +1,109 @@
+package obfuscator
+
+import (
+	"fmt"
+	"regexp"
+	"strings"
+	"sync"
+)
+
+type domainObfuscator struct {
+	ReplacementReporter
+	hostPattern          *regexp.Regexp
+	domainCount          int
+	lock                 sync.Mutex
+	tlds                 map[string]struct{}
+	mainDomains          map[string]string
+	ipPattern            *regexp.Regexp
+	fileExtensionPattern *regexp.Regexp
+}
+
+func (d *domainObfuscator) FileName(s string) string {
+	// if the filename has an extension omit it from obfuscation
+	// this is still error-prone because the last part of the domain name could still be mistaken for an extension
```

Comment:
isn't it a bit easier to take the `filepath.Ext` and just match on everything that comes before that? like: `s[: -len(filepath.Ext(s))]`. Would save a bit of regex matching

Reply:
Updated to use `filepath.Ext()`